composer/datasets/in_context_learning_evaluation.py

```diff
@@ -103,117 +108,171 @@ def _get_fewshot_sample_idxs(dataset_size: int, num_fewshot: int, sample_idx: in
+                                  remove_columns=dataset.column_names,
+                                  batched=batched,
+                                  fn_kwargs=hf_parsing_vars)
+
+        else:
+            with dist.local_rank_zero_download_and_wait(destination_path):
+                if dist.get_local_rank() == 0:
+                    get_file(dataset_uri, destination_path, overwrite=True)
+            dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
+        return dataset
+
+    def generate_few_shot_text(
+        self,
+        num_fewshot: int,
+        sample_idx: int,
+        preamble: str,
+        fewshot_rng: random.Random,
+    ) -> str:
         """Formats the prompt fewshot examples for test sample `sample_idx`.
 
         Randomly select `num_fewshot` samples from the dataset (not including the sample at `sample_idx`) and format
-        them each as follows `{example_delimiter}{question_prelimiter}{context}{continuation_delimiter}{chain_of_thought}{cot_delimiter}{answer}`.
+        them each as follows `{example_delimiter}{prelimiter}{context}{continuation_delimiter}{chain_of_thought}{cot_delimiter}{answer}`.
 
         `chain_of_thought` will default to empty if not present in the dataset but `context` and `answer` must be present.
 
         Returns the formatted prompt_string + concatenated list of formatted few shot examples.
         """
-        prompt_and_fewshot = prompt_string
+        few_shot_text = preamble
 
         if num_fewshot > 0:
-            fewshot_idxs = _get_fewshot_sample_idxs(len(self.samples), num_fewshot, sample_idx, fewshot_rng)
+            fewshot_idxs = _get_fewshot_sample_idxs(len(self.dataset), num_fewshot, sample_idx, fewshot_rng)
             for fewshot_idx in fewshot_idxs:
-                context = self.samples[fewshot_idx]['context']
-                chain_of_thought = self.samples[fewshot_idx].get('chain_of_thought', '')
-                answer = self.samples[fewshot_idx]['answer']
-
-                if len(chain_of_thought) == 0:
-                    cot_delimiter = ''
-                context = f'{question_prelimiter}{context}'
-                if len(prompt_and_fewshot) > 0:
-                    context = f'{example_delimiter}{context}'
-                prompt_and_fewshot += f'{context}{continuation_delimiter}{chain_of_thought}{cot_delimiter}{answer}'
-
-        return prompt_and_fewshot
-
-    def _prep_examples(self,
-                       num_fewshot: int,
-                       prompt_string: str,
-                       example_delimiter: str,
-                       continuation_delimiter: str,
-                       question_prelimiter: str,
-                       fewshot_rng: random.Random,
-                       cot_delimiter: str = '') -> List[Dict[str, Any]]:
+                ctxt = self.construct_context(self.dataset[fewshot_idx], few_shot_text, add_answer=True)
```

Comment:
where did CoT go?

Reply:
cot only exists in QA datasets AFAICT

Reply:
Do we want all datasets to have COT capabilities? I could fold the functionality into the default. If no cot exists, it would just get skipped. @bmosaicml opinions?